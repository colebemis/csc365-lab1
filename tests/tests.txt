// CSC 365. Spring 2019
// Lab 1-1 test suite

// TC-1
// Tests R3, R4
// short form command, existing student
// expected output: LINHART, LELA, 3, 107, FAFARD, ROCIO

S: LINHART

// TC-2
// Tests R3, R4
// long form command, existing student
// expected output: SWEDLUND, SHARRI, 3, 107, FAFARD, ROCIO

STUDENT: SWEDLUND

// TC-3
// Tests R3, R4
// short form command, non-existing student
// expected output: <empty line>

S: NEMO

// TC-4
// Tests R3, R4
// long form commands, missing student
// expected output:
//   Unrecognized command. Possible commands:
//   - S[tudent]: <lastname> [B[us]]
//   - T[eacher]: <lastname>
//   - B[us]: <number>
//   - G[rade]: <number> [H[igh]|L[ow]]
//   - A[verage]: <number>
//   - I[nfo]
//   - Q[uit]

STUDENT:

// TC-5
// Tests R3, R5
// short form commands, existing student
// expected output: TOWLEY, LANE, 54

S: TOWLEY B

// TC-6
// Tests R3, R6
// long form command, existing teacher
// expected output: 
//   RACANELLO, NOEL
//   CORONADO, DIMPLE
//   BOYTER, WAN
//   KEMERER, ROSETTA
//   DEMARTINI, DEWAYNE

TEACHER: KERBS

// TC-7
// Tests R3, R7
// short form command, existing grade
// expected output:
//   SAELEE, DANILO
//   GARTH, JOHN

G: 1

// TC-8
// Tests R3, R7
// long form command, non-existing grade
// expected output: <empty line>

GRADE: 17

// TC-9
// Tests R3, R8
// short form command, existing bus
// expected output:
//   COOKUS, XUAN, 3, 107
//   ELHADDAD, SHANTE, 3, 107
//   IMMERMAN, DEVIN, 4, 105
//   OBREIN, STAN, 6, 111
//   CORONADO, DIMPLE, 6, 102
//   SARAO, DIEDRA, 2, 108
//   EARLY, TORY, 3, 107
//   KEMERER, ROSETTA, 6, 102

B: 52

// TC-10
// Tests R3, R8, E1
// long form command, missing bus
// expected output:
//   Unrecognized command. Possible commands:
//   - S[tudent]: <lastname> [B[us]]
//   - T[eacher]: <lastname>
//   - B[us]: <number>
//   - G[rade]: <number> [H[igh]|L[ow]]
//   - A[verage]: <number>
//   - I[nfo]
//   - Q[uit]

BUS:

// TC-11
// Tests R3, R9
// long form command, existing grade
// expected output: GARTH, JOHN, 3.14, FALKER, ADOLPH, 0

GRADE: 1 h

// TC-12
// Tests R3, R9
// short form command, existing grade
// expected output: SAELEE, DANILO, 2.85, FALKER, ADOLPH, 54

G: 1 l

// TC-13
// Tests R3, R10
// short form command, existing grade
// expected output: 1: 2.995

A: 1

// TC-14
// Tests R3, R10, E1
// long form command, missing grade
// expected output:
//   Unrecognized command. Possible commands:
//   - S[tudent]: <lastname> [B[us]]
//   - T[eacher]: <lastname>
//   - B[us]: <number>
//   - G[rade]: <number> [H[igh]|L[ow]]
//   - A[verage]: <number>
//   - I[nfo]
//   - Q[uit]

AVERAGE:

// TC-15
// Tests R3, R11
// long form command
// expected output:
//   0: 0
//   1: 2
//   2: 13
//   3: 9
//   4: 15
//   5: 0
//   6: 21

INFO

// TC-16
// Tests R3, R12
// short form command
// expected output: program terminates

Q

// TC-17
// Tests E1
// expected output:
//   Unrecognized command. Possible commands:
//   - S[tudent]: <lastname> [B[us]]
//   - T[eacher]: <lastname>
//   - B[us]: <number>
//   - G[rade]: <number> [H[igh]|L[ow]]
//   - A[verage]: <number>
//   - I[nfo]
//   - Q[uit]

J: NONSENSE

// TC-18
// Tests R3, NR1
// short form command, existing classroom
// expected output:
//   RACANELLO, NOEL
//   CORONADO, DIMPLE
//   BOYTER, WAN
//   KEMERER, ROSETTA
//   DEMARTINI, DEWAYNE

C: 102

// TC-19
// Tests R3, NR2
// long form commands, existing classroom
// expected output: ALPERT, JONATHAN

CLASSROOM: 112 TEACHER
