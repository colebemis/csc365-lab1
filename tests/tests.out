// CSC 365. Spring 2019
// Lab 1-2 test suite
// Kellie Banzon, Cole Bemis, Tanner Larson

// TC-1
// Tests R3, R4z
LINHART, LELA, 3, 107, FAFARD, ROCIO

// TC-2
// Tests R3, R4
SWEDLUND, SHARRI, 3, 107, FAFARD, ROCIO


// TC-3
// Tests R3, R4


// TC-4
// Tests R3, R4
Unrecognized command. Possible commands:
- S[tudent]: <lastname> [B[us]]
- T[eacher]: <lastname>
- B[us]: <number>
- G[rade]: <number> [H[igh]|L[ow]]
- A[verage]: <number>
- I[nfo]
- Q[uit]

// TC-5
// Tests R3, R5
TOWLEY, LANE, 54

// TC-6
// Tests R3, R6
RACANELLO, NOEL
CORONADO, DIMPLE
BOYTER, WAN
KEMERER, ROSETTA
DEMARTINI, DEWAYNE

// TC-7
// Tests R3, R7
SAELEE, DANILO
GARTH, JOHN

// TC-8
// Tests R3, R7


// TC-9
// Tests R3, R8
COOKUS, XUAN, 3, 107
ELHADDAD, SHANTE, 3, 107
IMMERMAN, DEVIN, 4, 105
OBREIN, STAN, 6, 111
CORONADO, DIMPLE, 6, 102
SARAO, DIEDRA, 2, 108
EARLY, TORY, 3, 107
KEMERER, ROSETTA, 6, 102

// TC-10
// Tests R3, R8, E1
Unrecognized command. Possible commands:
- S[tudent]: <lastname> [B[us]]
- T[eacher]: <lastname>
- B[us]: <number>
- G[rade]: <number> [H[igh]|L[ow]]
- A[verage]: <number>
- I[nfo]
- Q[uit]

// TC-11
// Tests R3, R9
GARTH, JOHN, 3.14, FALKER, ADOLPH, 0

// TC-12
// Tests R3, R9
SAELEE, DANILO, 2.85, FALKER, ADOLPH, 54

// TC-13
// Tests R3, R10
2.995

// TC-14
// Tests R3, R10, E1
Unrecognized command. Possible commands:
- S[tudent]: <lastname> [B[us]]
- T[eacher]: <lastname>
- B[us]: <number>
- G[rade]: <number> [H[igh]|L[ow]]
- A[verage]: <number>
- I[nfo]
- Q[uit]

// TC-15
// Tests R3, R11
0: 0
1: 2
2: 13
3: 9
4: 15
5: 0
6: 21

// TC-16
// Tests R3, R12


// TC-17
// Tests E1
Unrecognized command. Possible commands:
- S[tudent]: <lastname> [B[us]]
- T[eacher]: <lastname>
- B[us]: <number>
- G[rade]: <number> [H[igh]|L[ow]]
- A[verage]: <number>
- I[nfo]
- Q[uit]
